jwst/ramp_fitting/ramp_fit.py

```diff
@@ -1086,72 +1246,84 @@ def fit_next_segment(start, end_st, end_heads, pixel_done, data_sect, mask_2d,
     #   semiramp (to enable unclassified pixels to have their arrays updated)
     got_case = np.zeros((asize1*asize2), dtype=np.bool)
 
-    # CASE A) Long enough (semiramp has >2 reads), at end of ramp
+    # CASE A) Long enough (semiramp has >2 groups), at end of ramp
     #    - set start to -1 to designate all fitting done
     #    - remove current end from end stack
     #    - set number of ends to 0
     #    - add slopes and variances to running sums
-    mask_sum = mask_2d.sum(axis=0)
-    wh_check = np.where((l_interval>2) & (end_locs==nreads - 1) & (mask_sum>0))
+    wh_check = np.where((l_interval>2) & (end_locs==nreads - 1) & (~pixel_done))
```

Comment:
Is it safe to use `&` for logical `and` with boolean arrays?  I can see that it works, but I wonder whether it's a throwback to when True and False were just aliases for 1 and 0.

Reply:
Yes, and this seems to be the recommended method. Using np.logical_and() also works.